sci-libs/opencascade/opencascade-7.7.1.ebuild

```
@@ -0,0 +1,366 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake virtualx
+
+DESCRIPTION="Development platform for CAD/CAE, 3D surface/solid modeling and data exchange"
+HOMEPAGE="https://www.opencascade.com"
+
+MY_PN="OCCT"
+
+MY_TEST_PV="7.7.0"
+MY_TEST_PV2="${MY_TEST_PV//./_}"
+
+SRC_URI="
+	test? ( https://github.com/Open-Cascade-SAS/${MY_PN}/releases/download/V${MY_TEST_PV2}/${PN}-dataset-${MY_TEST_PV}.tgz
+		-> ${PN}-dataset-${MY_TEST_PV}.tar.gz )
+"
+
+if [[ ${PV} = *9999* ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/Open-Cascade-SAS/${MY_PN}.git"
+else
+	MY_PV="${PV//./_}"
+	SRC_URI+="
+		https://github.com/Open-Cascade-SAS/${MY_PN}/archive/refs/tags/V${MY_PV}.tar.gz -> ${P}.tar.gz
+	"
+	S="${WORKDIR}/${MY_PN}-${MY_PV}"
+	KEYWORDS="~amd64 ~arm ~arm64 ~riscv ~x86"
+fi
+
+LICENSE="|| ( Open-CASCADE-LGPL-2.1-Exception-1.0 LGPL-2.1 )"
+SLOT="0/$(ver_cut 1-2)"
+IUSE="X debug doc examples ffmpeg freeimage freetype gles2-only +gui jemalloc json +opengl optimize tbb test testprograms tk vtk"
+
+REQUIRED_USE="
+	?? ( optimize tbb )
+	?? ( opengl gles2-only )
+	test? ( freeimage json opengl )
+"
+
+# There's no easy way to test. Testing needs a rather big environment properly set up.
+RESTRICT="!test? ( test )"
+
+# ffmpeg: https://tracker.dev.opencascade.org/view.php?id=32871
+RDEPEND="
+	dev-lang/tcl:=
+	tk? ( dev-lang/tk:= )
+	dev-libs/double-conversion
+	freetype? (
+		media-libs/fontconfig
+		media-libs/freetype:2
+	)
+	opengl? (
+		media-libs/libglvnd
+	)
+	X? (
+		x11-libs/libX11
+	)
+	gui? (
+		examples? (
+			dev-qt/qtcore:5
+			dev-qt/qtgui:5
+			dev-qt/qtquickcontrols2:5
+			dev-qt/qtwidgets:5
+			dev-qt/qtxml:5
+		)
+	)
+	ffmpeg? ( <media-video/ffmpeg-5:= )
+	freeimage? ( media-libs/freeimage )
+	jemalloc? ( dev-libs/jemalloc )
+	tbb? ( dev-cpp/tbb:= )
+	vtk? (
+		sci-libs/vtk:=[rendering]
```

```suggestion
		sci-libs/vtk:=[opencascade,rendering]
```
``>=sci-libs/vtk-9.3.0`` require ``opencascade`` use flag